### PR TITLE
add new api to receive events on k8s config change

### PR DESCRIPTION
**Pull request by evidolob**
This PR add two event for external API which allow to receive notifications when `vscode-kubernetes-tools` change path to k8s config file or change active cluster.

This PR will help external extensions to detect and react on changes config file in one consistent way. Previously the only way to detect active cluster change was adding own file system watcher on default k8s config path. But this will not work, if user provide custom config file, which can be placed on any place of filesystem.

### CODE REVIEW

**Comment by evidolob** [comment]
@itowlson Thx for review. Yeh, I'm not very good on writing docs. So if you can help with them...

Another thing, that I noticed, there https://github.com/Azure/vscode-kubernetes-tools/pull/766/files#diff-2c673f0f57a051f275eec6d5e4e03e24R17 documentation is not actual, 
according to current code `getKubeconfigPath` function return an object not a `Promise`.

**Comment by evidolob** [comment]
@lstocchi I add event for k8s namespace change. But I think it not cover all https://github.com/Azure/vscode-kubernetes-tools-api/issues/19 requests. As this PR add ability to track namespace change but not provide the way to get current namespace.

File: src/api/implementation/configuration/versions.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+import * as v1_1 from "./v1_1";
 import { API } from "../../contract/api";
 import { versionUnknown, available } from "../apiutils";
+import { EventEmitter } from "vscode";
+import { ActiveValueTracker } from "../../../components/contextmanager/active-value-tracker";
+import { ConfigurationV1 } from "../../contract/configuration/v1";
 
-export function apiVersion(version: string): API<any> {
+export function apiVersion(version: string,
+    onDidChangeKubeconfigEmitter: EventEmitter<ConfigurationV1.KubeconfigPath>,
```
Comment (lstocchi) [inline src/api/implementation/configuration/versions.ts:12]: It works the same but i would suggest to use the KubeconfigPath from `kubectl/kubeconfig`

```
import { KubeconfigPath } from "../../../components/kubectl/kubeconfig"; 

onDidChangeKubeconfigEmitter: EventEmitter<KubeconfigPath>
```

**Comment by lstocchi** [comment]
> @lstocchi I add event for k8s namespace change. But I think it not cover all [Azure/vscode-kubernetes-tools-api#19](https://github.com/Azure/vscode-kubernetes-tools-api/issues/19) requests. As this PR add ability to track namespace change but not provide the way to get current namespace.

Ok, don't worry @evidolob. We can also add it later :) I just mentioned that issue because i think it was a bit related to this patch but we can work on that another time. Thanks again for the namespace change listener, it works great and i think it's really useful!! :+1:

File: src/components/kubectl/namespace.ts [hunk 2]
```diff
@@ -6,12 +6,16 @@ import * as kuberesources from '../../kuberesources';
...
+import { EventEmitter } from 'vscode';
+
+export const onDidChangeNamespaceEmitter = new EventEmitter<string>();
 
 export async function useNamespaceKubernetes(kubectl: Kubectl, explorerNode: ClusterExplorerNode) {
     if (explorerNode && explorerNode.nodeType === NODE_TYPES.resource) {
         if (await kubectlUtils.switchNamespace(kubectl, explorerNode.name)) {
             refreshExplorer();
             host.showInformationMessage(`Switched to namespace ${explorerNode.name}`);
+            onDidChangeNamespaceEmitter.fire(explorerNode.name);
```
[the remark this answers is not in the document]
Comment (evidolob) [reply src/components/kubectl/namespace.ts]: Yes, I miss that `kubectlUtils.switchNamespace` was called not only from `useNamespaceKubernetes `. Done

**Comment by evidolob** [comment]
@lstocchi I just want highlight that this PR do not solve Azure/vscode-kubernetes-tools-api#19 completely.

**Comment by lstocchi** [comment]
@itowlson updated the documentation a bit. WDYT?

**Comment by evidolob** [comment]
@itowlson ping, WDYT?

**Comment by lstocchi** [comment]
@bnookala Can you give it a look at this? It should be ready

**Comment by lstocchi** [comment]
@itowlson could you please give this a look? It would be great if we could include this patch in the next release.

**Review by itowlson — APPROVED** [review]
This looks good - thanks heaps!  Sorry for the delay.

**Pull request merged**